newrelic_lambda_cli/cli/layers.py

```diff
@@ -94,54 +95,24 @@ def register(group):
     help="Enable/disable sending Lambda function logs via the Extension",
 )
 @click.pass_context
-def install(
-    ctx,
-    nr_account_id,
-    nr_api_key,
-    nr_region,
-    aws_profile,
-    aws_region,
-    aws_permissions_check,
-    functions,
-    excludes,
-    layer_arn,
-    upgrade,
-    enable_extension,
-    enable_extension_function_logs,
-):
+def install(ctx, **kwargs):
     """Install New Relic AWS Lambda Layers"""
-    session = boto3.Session(profile_name=aws_profile, region_name=aws_region)
-
-    if aws_permissions_check:
-        permissions.ensure_lambda_install_permissions(session)
-
-    functions = get_aliased_functions(session, functions, excludes)
-
-    def config_layers(function):
-        result = layers.install(
-            session,
-            function,
-            layer_arn,
-            nr_account_id,
-            nr_api_key,
-            nr_region,
-            upgrade,
-            enable_extension,
-            enable_extension_function_logs,
-            ctx.obj["VERBOSE"],
+    input = LayerInstall(session=None, verbose=ctx.obj["VERBOSE"], **kwargs)
+
+    input = input._replace(
+        session=boto3.Session(
+            profile_name=input.aws_profile, region_name=input.aws_region
         )
+    )
+
+    if input.aws_permissions_check:
+        permissions.ensure_layer_install_permissions(input)
 
-        if result and enable_extension_function_logs:
-            subscriptions.remove_log_subscription(session, function)
-        return result
+    functions = get_aliased_functions(input)
 
     with ThreadPoolExecutor() as executor:
         futures = [
-            executor.submit(
-                config_layers,
-                function,
-            )
-            for function in functions
+            executor.submit(layers.install, input, function) for function in functions
```

Comment:
Wait, we got rid of the config_layers closure. It looks like we're not removing the subscription filter anymore.

Reply:
@MattWhelan I moved the subscription filter code to the same place where we check if the extension is enabled and add the license key policy. Trying to keep the logic in the CLI bits to a minimum.

Reply:
Code was moved here: https://github.com/newrelic/newrelic-lambda-cli/pull/139/files#diff-d14719dcdcf446bd00c4b7977b0acee473aa1e5dbce20b230ccca5e856e3c9cfR191